packages/react-native-web/src/exports/Image/index.js

```diff
@@ -332,14 +343,76 @@ const Image: React.AbstractComponent<
   );
 });
 
-Image.displayName = 'Image';
+BaseImage.displayName = 'Image';
+
+/**
+ * This component handles specifically loading an image source with headers
+ * default source is never loaded using headers
+ */
+const ImageWithHeaders: ImageComponent = React.forwardRef((props, ref) => {
+  // $FlowIgnore: This component would only be rendered when `source` matches `ImageSource`
```

Comment:
I'm a bit confused by this comment - isn't `ImageSource` just a type? And is 'source' in the comment 'props.source'?

I think you're referring to `if (props.source && props.source.headers) {` check which will render this, component down below, right?

Reply:
> I think you're referring to `if (props.source && props.source.headers) {` check which will render this, component down below, right?

Yes, I'm referring to the prior check
I'm trying to explain that `source` would always be specific subset of the `Source` type which is `ImageSourde` - an object with headers

There's probably a `flow` way to declare this
like making `ImageWithHeaders` use different props, particularly a `source` prop of type `ImageSource` 

I can change that to `$FlowFixMe` if you want to

Reply:
Cool cool nah let's keep it 👍